storage/src/main/java/tech/pegasys/teku/storage/client/RecentChainData.java

```diff
@@ -186,6 +187,12 @@ public UpdatableStore getStore() {
         .orElseGet(TreeMap::new);
   }
 
+  public NavigableMap<UInt64, Bytes32> getAncestorRootsForRoot(
+      final UInt64 startSlot, Bytes32 root) {
+    return ForkChoiceUtil.getAncestors(
+        forkChoiceStrategy.orElseThrow(), root, startSlot, UInt64.ONE, UInt64.valueOf(10000));
```

Comment:
Hard coded 10000 doesn't look right here.

Reply:
Also I'm suspicious that we might be including the common ancestor block itself when processing a reorg.  We should process both chains only from the block after the common ancestor (ie only process the blocks that actually changed).

Reply:
Yeah, I wasn't sure about 10000 as well. I need a number high enough to never be a problem, but less than the limits of UInt64 so that it never throws an `ArithmeticException` as well.

Reply:
Added a new method to get every ancestor that does not calculate `count` instead of passing a magic variable.

Reply:
Also made sure this method does not return the block root for the common ancestor.